utils/report.c

```diff
@@ -81,6 +81,101 @@ static struct uftrace_report_node *find_or_create_node(struct rb_root *root, con
 	return node;
 }
 
+struct call_stat {
+	uint64_t addr;
+	struct rb_node link;
+	uint64_t callers[1024];
+	uint64_t callees[1024];
+	int caller_count;
+	int callee_count;
+};
+
+static struct rb_root call_stat_root = RB_ROOT;
```

Comment:
But now I wonder why we need this separate `call_stat` structure.  Can't we add it to `uftrace_report_node`?

Reply:
@namhyung 
I just thought implementation could be complex if `call_stat` is located in `uftrace_report_node`.
I should add some arguments for `call_stat` such as `addr` and `parent_node` in `find_or_create_node()`. Then I should change its callers' arguments and so on. 

But if you think relocating `call_stat` to uftrace_report_node` is better, I'm going to change it.

Reply:
I think we can rename (and add) something like `get_node_by_name()` and `get_node_by_addr()`.  Having a separate struct might be more confusing and error-prone.